frameworklauncher/src/main/java/com/microsoft/frameworklauncher/webserver/LauncherModule.java

```diff
@@ -65,15 +68,89 @@ private static LaunchClientType getLaunchClientType(
     }
   }
 
-  private static String getUserName(
-      CommonExts.NoExceptionCallable<String> ResolveUserName) throws BadRequestException {
-    String userName = ResolveUserName.call();
-    if (userName == null) {
+  private static String getName(
+      CommonExts.NoExceptionCallable<String> ResolveName) throws BadRequestException {
+    String name = ResolveName.call();
+    if (name == null) {
       return null;
     }
 
-    CommonValidation.validate(userName);
-    return userName;
+    CommonValidation.validate(name);
+    return name;
+  }
+
+  private static Boolean getBoolean(
+      CommonExts.NoExceptionCallable<String> ResolveBoolean) throws BadRequestException {
+    String booleanStr = ResolveBoolean.call();
+    if (booleanStr == null) {
+      return null;
+    }
+
+    if (!booleanStr.equalsIgnoreCase("true") && !booleanStr.equalsIgnoreCase("false")) {
+      throw new BadRequestException(String.format(
+          "Failed to ParseBooleanStr: [%s]", booleanStr));
+    }
+
+    return Boolean.valueOf(booleanStr);
+  }
+
+  private void checkWritableAccess(
+      HttpServletRequest hsr) throws Exception {
+    checkWritableAccess(hsr, null, null);
+  }
+
+  private void checkWritableAccess(
+      HttpServletRequest hsr,
+      String frameworkName) throws Exception {
+    checkWritableAccess(hsr, frameworkName, null);
+  }
+
+  private void checkWritableAccess(
+      HttpServletRequest hsr,
+      String frameworkName,
+      UserDescriptor frameworkUser) throws Exception {
+    if (!conf.getWebServerAclEnable()) {
+      return;
+    }
+
+    // Check Admin
+    Boolean isAdmin = getBoolean(() -> hsr.getHeader(WebCommon.REQUEST_HEADER_IS_ADMIN));
+    if (isAdmin != null && isAdmin) {
+      // Admin Users can always access anything
+      return;
+    } else if (frameworkName == null) {
+      // Non Admin can only write Framework Request
+      throw new AuthorizationException(
+          "This operation needs administrator privilege.");
+    }
+
+    UserDescriptor user = UserDescriptor.newInstance(
+        getName(() -> hsr.getHeader(WebCommon.REQUEST_HEADER_USER_NAME)));
+
+    // Check UserName consistency
+    if (frameworkUser != null && !frameworkUser.equals(user)) {
+      throw new AuthorizationException(String.format(
+          "User [%s] cannot submit a Framework which belongs to another User [%s].",
+          user, frameworkUser));
```

Comment:
>frameworkUser [](start = 16, length = 13)

It not privacy to expose internal user to other user.

Reply:
frameworkUser is the user write in his description....

---
In reply to: [167495024](https://github.com/Microsoft/pai/pull/320#discussion_r167495024) [](ancestors = 167495024)

Reply:
This is for data deployment to set user. 
Anyway, I will make it more clear.

---
In reply to: [167745469](https://github.com/Microsoft/pai/pull/320#discussion_r167745469) [](ancestors = 167745469,167495882,167495024)